### PR TITLE
add alibaba druid support more database tracing sql query

### PR DESCRIPTION
**Pull request by IAMTJW**
This is a tracing filter for Alibaba Druid 1.1.14 (https://github.com/alibaba/druid), support more databases。

### CODE REVIEW

**Comment by adriancole** [comment]
@IAMTJW maybe you can post this to your QQ group to see if others want this integration? If we have more thumbsup here we can proceed. https://github.com/alibaba/druid/wiki/FAQ#23-questions-and-suggestions-are-welcome

File: instrumentation/alibaba-druid/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# brave-instrumentation-alibaba-druid
+
+## Druid Filter
+
+This is a tracing filter for  [Alibaba Druid 1.1.14 ](https://github.com/alibaba/druid)，support more databases(mysql、postgresql、oracle、Microsoft、DB2、sqlite etc.) to tracing sql query and can replace brave-instrumentation-mysql.
+
+## Configuration
+
+    <bean id="tracingStatementFilter" class="brave.druid.TracingStatementFilter">
```
Comment (adriancole) [inline instrumentation/alibaba-druid/README.md]: wrap in ```xml and please format so indents are consistent

File: instrumentation/alibaba-druid/pom.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        <dependency>
+            <groupId>com.alibaba</groupId>
+            <artifactId>druid</artifactId>
+            <version>1.1.14</version>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
```
Comment (adriancole) [inline instrumentation/alibaba-druid/pom.xml]: probably choose one in-memory for test scope, such as derby. you can look at p6spy for an example.

File: instrumentation/alibaba-druid/src/main/java/brave/druid/TracingStatementFilter.java [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+    }
+
+
+    @Override
+    protected void statementExecuteUpdateBefore(StatementProxy statement, String sql) {
+
+        super.statementExecuteUpdateBefore(statement,sql);
+
+        try {
```
Comment (adriancole) [inline instrumentation/alibaba-druid/src/main/java/brave/druid/TracingStatementFilter.java]: there are some concerns in this file. I can see you are doing some defensive code, but possibly there are ways to do this in less code. for example, if your before and after blocks did the defense, the delegation methods would not need to.

File: instrumentation/alibaba-druid/pom.xml (hunk 2 above)
Comment (IAMTJW) [reply instrumentation/alibaba-druid/pom.xml]: thanks

File: instrumentation/alibaba-druid/src/main/java/brave/druid/TracingStatementFilter.java (hunk 3 above)
Comment (IAMTJW) [reply instrumentation/alibaba-druid/src/main/java/brave/druid/TracingStatementFilter.java]: ok

File: instrumentation/alibaba-druid/README.md (hunk 1 above)
Comment (IAMTJW) [reply instrumentation/alibaba-druid/README.md]: thanks

**Comment by wangrangzhi** [comment]
wonderful support for sql！

**Comment by adriancole** [comment]
I currently see two people interested in this. We need to see more to justify the review of this change and future maintenance.

**Comment by adriancole** [comment]
in over one year only 2 people say interest. Unless we hear more I will close at the end of the week. @IAMTJW sorry just we have to make sure we use time maintaining things that a lot of people use.

**Comment by jorgheymans** [comment]
If really still interested they could also develop and maintain it as a standalone module in https://github.com/openzipkin-contrib perhaps.

**Comment by adriancole** [comment]
agree @jorgheymans , but even there should be at least 3 people I think as setting up the project is even more work than another module here.

**Pull request closed without merging**